DANS-KNAW/dans-datastation-tools
# DD-1493: Add command for dataverse role assignment

Fixes DD-1493

# Description of changes
Added dv-dataverse-role-assignment, similar to that for datasets, but needs aliases instead of PIDs

# How to test
Assuming you have an assigment like the following; a mail domain for `exampleorg` and a dataverse with the alias  `dans`:

`poetry run dv-dataverse-role-assignment list dans`
Should not list that assignment.

`poetry run dv-dataverse-role-assignment add '&maildomain/exampleorg=dsContributor' dans`
Should be stating it was added

`poetry run dv-dataverse-role-assignment add '&maildomain/exampleorg=dsContributor' dans`
Should be stating it wasn't needed, because it was already assigned. 

`poetry run dv-dataverse-role-assignment list dans`
Should list that assignment.

`poetry run dv-dataverse-role-assignment remove '&maildomain/exampleorg=dsContributor' dans`
Should be stating it was removed

`poetry run dv-dataverse-role-assignment remove '&maildomain/exampleorg=dsContributor' dans`
Should be stating it wasn't needed, because it was not assigned.
 
`poetry run dv-dataverse-role-assignment list dans`
Should not list that assignment.

The add and remove commands should be run with a file containing several aliases of dataverses. Producing similar results, but then for all the aliases. 

Because of the refactoring it would be wise to also test `dv-dataverse-root-collect-permission-overview` and `dv-dataverse-root-collect-storage-usage`. 

# Related PRs

(Add links)

*https://github.com/DANS-KNAW/dans-datastation-tools/pull/52

# Notify

@DANS-KNAW/core-systems
